### PR TITLE
Bill/misc documentation

### LINKED ISSUE #102: Miscellaneous Information
Add documents for things to note about STM Board

### PR DESCRIPTION
**Pull request by bill-luu**
Fixes #102 

Also, moved the STM information inside the Development Resources to here too, as I think it makes more sense to have it here.

### CODE REVIEW

File: Documentation/CompileFlashSTMBoard.adoc [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+If a board isn't connected, and it ran successfully, the output should be: 
+----
+st-flash 1.3.0-dev
+DATE/TIME WARN /opt/stlink/src/usb.c: Couldn't find any ST-Link/V2 devices
+----
+
+=== Black Magic Probe
+
+The Black Magic Probe is compatable both the 10-Pin & 20-Pin JTag headers.
```
Comment (safasyed) [inline Documentation/CompileFlashSTMBoard.adoc]: compatible*

File: Documentation/CompileFlashSTMBoard.adoc [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+. Load the program into the board. `load`
+
+. Breakpoints can be set to help debug any potential issues. See the https://www.gnu.org/software/gdb/documentation/[official gdb guide] for more useful commands.
+
+
+Alternatively, this can be done in one command by attaching mulitple `-ex` flags to the `arm-none-eabi-gdb` command:
+
+```
+ sudo arm-none-eabi-gdb EpsilonCCS.elf -ex 'target extended-remote /dev/ttyACM0' -ex 'monitor tpwr enable' -ex 'monitor swdp_scan' -ex 'attach 1' -ex 'load'
+
```
Comment (safasyed) [inline Documentation/CompileFlashSTMBoard.adoc]: add that there is also a script to flash with gdb and how to use it

File: Documentation/CompileFlashSTMBoard.adoc [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+. Make note of the Toolchain Folder Location, which will also be the project folder location.
+
+. Click OK and download any dependencies that are needed, then click OK again.
+
+. Make pin assignments then go to Project > Generate Code, and then save project.
+
+
+== Generating a Makefile and Compiling
+
+. Create the makefile using  `CubeMX2Makefile /path/to/project/folder`.
```
[the remark this answers is not in the document]
Comment (bill-luu) [reply Documentation/CompileFlashSTMBoard.adoc]: This can be done in any directory

File: Documentation/CompileFlashSTMBoard.adoc [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+If a board isn't connected, and it ran successfully, the output should be: 
+----
+st-flash 1.3.0-dev
+DATE/TIME WARN /opt/stlink/src/usb.c: Couldn't find any ST-Link/V2 devices
+----
+
+=== Black Magic Probe
+
+The Black Magic Probe is compatible both the 10-Pin & 20-Pin JTag headers.
```
Comment (avneetgill) [inline Documentation/CompileFlashSTMBoard.adoc]: compatible with*

**Pull request merged**